libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -541,7 +542,11 @@ TEST( TEST_IOT_UART, AFQP_IotUARTCancel )
 
         /* Wait to make sure operation was really canceled. */
         xCallbackReturn = xSemaphoreTake( xWriteCompleteSemaphore, testIotUART_DEFAULT_SEMPAHORE_DELAY );
-        TEST_ASSERT_EQUAL( pdFALSE, xCallbackReturn );
+
+        lCancel = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lCancel );
+
+        TEST_ASSERT_TRUE( pdFALSE == xCallbackReturn || lTransferAmount < testIotUART_BUFFER_LENGTH );
```

Comment:
IMO: I think this is should be two separate asserts, it may be difficult to debug which case failed in this assert.